psalm-baseline.xml

```diff
@@ -258,6 +274,47 @@
       <code>new static()</code>
     </UnsafeInstantiation>
   </file>
+  <file src="src/Generator/EnumGenerator/Cases/CaseFactory.php">
+    <ArgumentTypeCoercion occurrences="1">
+      <code>$pureCases</code>
+    </ArgumentTypeCoercion>
+    <MixedArgument occurrences="2">
+      <code>$backingType-&gt;getName()</code>
+      <code>$enum-&gt;getCases()</code>
+    </MixedArgument>
+    <MixedArgumentTypeCoercion occurrences="1">
+      <code>$backedCases</code>
+    </MixedArgumentTypeCoercion>
+    <MixedArrayOffset occurrences="1">
+      <code>$backedCases[$singleCase-&gt;getName()]</code>
+    </MixedArrayOffset>
+    <MixedAssignment occurrences="3">
+      <code>$backedCases[$singleCase-&gt;getName()]</code>
+      <code>$backingType</code>
+      <code>$singleCase</code>
+    </MixedAssignment>
+    <MixedMethodCall occurrences="3">
+      <code>getBackingValue</code>
+      <code>getName</code>
+      <code>getName</code>
+    </MixedMethodCall>
+    <UndefinedClass occurrences="1">
+      <code>ReflectionEnumUnitCase</code>
+    </UndefinedClass>
+    <UndefinedMethod occurrences="3">
+      <code>getBackingType</code>
+      <code>getCases</code>
+      <code>getCases</code>
+    </UndefinedMethod>
+  </file>
+  <file src="src/Generator/EnumGenerator/EnumGenerator.php">
+    <MixedArgument occurrences="1">
+      <code>$enum-&gt;getName()</code>
+    </MixedArgument>
+    <UndefinedMethod occurrences="1">
+      <code>getName</code>
+    </UndefinedMethod>
+  </file>
```

Comment:
Not sure if worth to do but you could extend the polyfill to get rid of this.

Reply:
IMO not worth doing here - these will disappear as we increase minimum PHP version, when viable.